python-frontend/src/test/java/org/sonar/python/semantic/ProjectLevelSymbolTableTest.java

```diff
@@ -987,6 +987,59 @@ void django_views() {
     assertThat(bar.isDjangoView()).isFalse();
   }
 
+  @Test
+  void django_views_local_functions() {
+    String content = """
+      from django.urls import path
+      
+      def foo(): ...
+      urlpatterns = [path('foo', foo, name='foo')]
+      
+      class MyClass:
+        def bar(): ...
+      
+      urlpatterns.append(path('bar', MyClass.bar, name='bar'))
+      
+      class MyOtherClass:
+        class MyNestedClass:
+          def qix(): ...
+      urlpatterns.append(path('bar', MyOtherClass.MyNestedClass.qix, name='bar'))
+      """;
+
+    ProjectLevelSymbolTable projectSymbolTable = new ProjectLevelSymbolTable();
+    projectSymbolTable.addModule(parseWithoutSymbols(content), "my_package", pythonFile("urls.py"));
+    assertThat(projectSymbolTable.isDjangoView("my_package.urls.foo")).isTrue();
+    assertThat(projectSymbolTable.isDjangoView("my_package.urls.MyClass.bar")).isTrue();
+    assertThat(projectSymbolTable.isDjangoView("my_package.urls.MyOtherClass.MyNestedClass.qix")).isTrue();
+  }
+
+  @Test
+  void django_views_ambiguous() {
+    String content = """
+      from django.urls import path
+      if x:
+        def ambiguous(): ...
+      else:
+        def ambiguous(): ...
+      urlpatterns.append(path('bar', ambiguous, name='bar'))
```

Comment:
Minor: here is definition of the `urlpatterns` is missed which could confuse reading this test case